Sources/Widgets/Widgets3D/ShapeWidget/behavior.js

```diff
@@ -498,7 +498,7 @@ export default function widgetBehavior(publicAPI, model) {
       publicAPI.invokeInteractionEvent();
     }
 
-    return model.hasFocus ? macro.EVENT_ABORT : macro.VOID;
+    return model.hasFocus || model.isDragging ? macro.EVENT_ABORT : macro.VOID;
```

Comment:
@floryst Shouldn't the widget get "focus" when its handle is being dragged ? Apparently widgets get focus only at creation-time. Is it on purpose ? When should they have "focus" ? Thanks !

Reply:
"Focus" as a widget concept means that a particular widget should be the only one interactable. This use-case arises when there is a primary widget.

An example is the paint widget needs focus, since when it's active you shouldn't be able to move rulers, crosshairs, etc. The widget manager (should) only allow the paint state to be active, and not activate any other state.

When dragging, that particular widget state is considered active, but the widget is not focused. Conversely, focus usually implies some state is active (like in the paint widget case).

Reply:
Thanks for your explanation ! it makes sense.

@tomsuchel please update the documentation with Forrest's words.